htdocs/salaries/class/paymentsalary.class.php

```diff
@@ -228,11 +228,10 @@ public function create($user, $closepaidcontrib = 0)
 
 		$totalamount = 0;
 		foreach ($this->amounts as $key => $value) {  // How payment is dispatch
-			$newvalue = price2num($value, 'MT');
+			$newvalue = (float) price2num($value, 'MT');
 			$this->amounts[$key] = $newvalue;
 			$totalamount += $newvalue;
 		}
-		$totalamount = price2num($totalamount);
```

Comment:
totalamount is result of += of a float value.
So result is a float. So may be a 0 may be stored as 0.0000000000001
This means the test just after 
`if ($totalamount == 0) 
`will fails when it should be true.
A price2num should solve this but it will work if there is a rounding param. So i suggest to add here
`$totalamount = price2num($totalamount, 'MT');`
instead of removing it completely...